### DISCUSSION TITLE
Error: "Type '(num: number) => number' is not assignable to type 'number'.

### DISCUSSION THREAD

**Discussion by gold-pudge** [community/community#39382] [Projects and Issues] [answered]
### Select Topic Area

Bug

### Body

When I have a function like this, I get no complaints:

const roundToTwo = (num: number) => {
  return +(Math.round(+(num + "e+2")) + "e-2");
};

**Comment by codpen** [comment, marked as answer]
roundToTwo is a function type, not a number
Given this defintion:

const roundToTwo = (num: number) => {
  return +(Math.round(+(num + "e+2")) + "e-2")
}
this is the actual (and correctly inferred) type of roundToTwo:

(num: number) => number
Why? Because roundToTwo is a [function type](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-type-expressions), not a number. The type is inferred from the function expression being assigned to it. Remember, in Javascript, functions are first class objects. Likewise, in Typescript they are first class types.

**Reply by gold-pudge** [reply]
Thank you for your help
